Content.Server/StationEvents/Events/MouseMigrationRule.cs

```diff
@@ -1,19 +1,39 @@
 using System.Linq;
 using Content.Server.GameTicking.Rules.Components;
+using Content.Server.Station.Components;
+using Content.Server.Station.Systems;
 using Content.Server.StationEvents.Components;
 using Robust.Shared.Random;
+using Robust.Shared.Utility;
 
 namespace Content.Server.StationEvents.Events;
 
 public sealed class MouseMigrationRule : StationEventSystem<MouseMigrationRuleComponent>
 {
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+
     protected override void Started(EntityUid uid, MouseMigrationRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
         base.Started(uid, component, gameRule, args);
 
+        var targetStation = _stationSystem.GetStations().FirstOrNull();
+
+        if (!TryComp(targetStation, out StationDataComponent? data))
+        {
+            Logger.Info("TargetStation not have StationDataComponent");
+            return;
+        }
```

Comment:
1. This should be targeting any station, not the first.
2. This is just a Resolve.

Reply:
I don't really understand what the second paragraph means.

Reply:
Use Resolve and pass in the datacomponent if you want to check if it exists and log for it.